### ISSUE TITLE
Crash in pcap_activate() on Linux

### ISSUE THREAD

**Issue by IanSMiller** [the-tcpdump-group/libpcap#1222]
The precise crash varies I have seen:
"free(): double free detected in tcache 2" and segment fault. When i occurs it is entirely reproducable.

It only seems to occur (at least I have only provoked it) where the executable has been granted the CAP_NET_RAW capability but has not been granted the CAP_NET_ADMIN capability.
I have reproduced the problem with the following code:-

```
#include <pcap/pcap.h>
#include <stdio.h>

int main(int argc, char * argv[])
{
    (void)argc;
    (void)argv;
    char errbuf[PCAP_ERRBUF_SIZE] = { 0 };
    if (pcap_init(PCAP_CHAR_ENC_UTF_8, errbuf))
    {
        fprintf(stderr, "pcap_init failed with %s\n", errbuf);
    }
    else
    {
        pcap_t * result = pcap_create("enp0s31f6", errbuf);
        if (!result)
        {
            fprintf(stderr, "pcap_create failed with %s\n", errbuf);
        }
        else
        {
            int ret;
            if (0 != (ret = pcap_set_tstamp_type(result, PCAP_TSTAMP_ADAPTER_UNSYNCED)))
                fprintf(stderr, "pcap_set_tstamp_type failed with %d\n", ret);
            else
            {
                printf("About to call pcap_activate\n");
                if (pcap_activate(result))
                    fprintf(stderr, "pcap_activate failed\n");
                else
                    fprintf(stderr, "pcap_activate succeeded\n");
            }
        }
    }
    return 1;
}
```

Where the above source is in file bug.c, the program is built with:
`g++ -o bug bug.cpp -lpcap -lstdc++
sudo setcap CAP_NET_RAW=ep bug`
When this is run it produces:
`About to call pcap_activate
Segmentation fault
`
I am using:-
libpcap version 1.10.0 (with TPACKET_V3)
Linux  5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux
CPU: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
compiler: g++ (Debian 10.2.1-6) 10.2.1 20210110
NIC: Intel Ethernet Connection (5) I219-LM (this the "enp0s31f6" referenced in the pcap_create call)
(It seems to be necessary to request PCAP_TSTAMP_ADAPTER_UNSYNCED timestamping to provoke the problem.  I cannot reproduce this with the machine's other NIC as it does not support this timestamping.)

The obvious get-round is to grant the CAP_NET_ADMIN.

**Comment by guyharris** [comment]
Duplicate of #1054.  You need a system that provides a later version of libpcap than 1.10.0; the current release is 1.10.4, but the fix, commit 892212ee18f0b45b56b19d58a0da55fa119e4a8d, is, I think, in 1.10.2 and later.

> The obvious get-round is to grant the CAP_NET_ADMIN.

As far as I know, that's not a workaround, that's a *requirement* for using non-host time stamps on Linux.

**Comment by guyharris** [comment]
Duplicate of #1054.
